apps/ae_socket_connector/lib/session_holder.ex

```diff
@@ -8,18 +8,42 @@ defmodule SessionHolder do
             color: nil,
             network_id: nil,
             priv_key: nil,
-            socket_connector_state: %SocketConnector{}
+            file: nil,
+            socket_connector_state: nil,
+            connection_callbacks: nil
 
   def start_link(%{
-        socket_connector: %SocketConnector{} = socket_connector_state,
+        socket_connector: socket_connector_state,
+        log_config: log_config,
         ae_url: ae_url,
         network_id: network_id,
         priv_key: priv_key,
+        connection_callbacks: connection_callbacks,
         color: color,
         # pid name, of the session holder, which is maintined over re-connect/re-establish
         pid_name: name
       }) do
-    GenServer.start_link(__MODULE__, {socket_connector_state, ae_url, network_id, priv_key, color}, name: name)
+    log_path = Map.get(log_config, :log_path, "log")
```

Comment:
This path should be called `log`, but `data`. It contains data files.

Reply:
Did some attempt to address this. Confusing. Are you fine with _logging_ data to a data folder

Reply:
fixed

Reply:
Fixed